paper-api/src/main/java/io/papermc/paper/world/biome/BiomeClimate.java

```diff
@@ -0,0 +1,35 @@
+package io.papermc.paper.world.biome;
+
+import org.bukkit.Location;
+
+public interface BiomeClimate {
+    /**
+     * Determines if the biome has precipitation.
+     *
+     * @return true if the biome has precipitation.
+     */
+    boolean hasPrecipitation();
+
+    /**
+     * Controls gameplay features like grass and foliage color,
+     * and a height adjusted temperature.
+     *
+     * @return the temperature of the biome.
+     */
+    float temperature();
+
+    /**
+     * Controls grass and foliage color.
+     *
+     * @return the downfall of the biome.
+     */
+    float downfall();
+
+    /**
+     * Modifies temperature before calculating the height adjusted temperature.
```

Comment:
The "modifies temperature" part confused me a bit until I looked at the actual implementation, so I feel it's worth clarifying.

```suggestion
     * Calculates the temperature based on climate and height.
```